### DISCUSSION TITLE
expand and collapse function

### DISCUSSION THREAD

**Discussion by lucienlmy** [dail8859/NotepadNext#448] [Q&A]
Why does my notepad next not have the expand and collapse function in the xml document?

**Comment by dail8859** [comment]
Can confirm folding does not work for XML. I think I have found the problem and it is an easy fix once I have time. Will create an issue from this discussion.

**Reply by lucienlmy** [reply]
What are the supported document types with folding function in this version? Can you add this function?

**Reply by dail8859** [reply]
Folding is dependent on the 3rd party library [Lexilla](https://github.com/ScintillaOrg/lexilla). Most (but not all) languages support folding. There are cases where it's supported but they required a specific setting to enable it, which is why many languages (python, C++, C#, etc) had folding by default, but XML/HTML required another tweak to get it working from the application side.

**Comment by ErickWei** [comment]
need support on this issue,come on

**Reply by dail8859** [reply]
Nobody will be able to help you if you do not provide any details. XML folding has already been fixed.

**Reply by ErickWei** [reply]
sorry and I've created a new issue describing this request, thanks
#526
